crates/forge/src/main.rs

```diff
@@ -111,71 +112,73 @@ fn main_execution() -> Result<bool> {
         .match_many(&scarb_metadata)
         .context("Failed to find any packages matching the specified filter")?;
 
-    let package_root = &scarb_metadata.workspace.root;
+    let workspace_root = &scarb_metadata.workspace.root;
 
     if args.clean_cache {
-        clean_cache(package_root).context("Failed to clean snforge cache")?;
+        clean_cache(workspace_root).context("Failed to clean snforge cache")?;
     }
 
-    let mut all_failed_tests = vec![];
-    for package in &packages {
-        let forge_config = config_from_scarb_for_package(&scarb_metadata, &package.id)?;
-        let (package_path, package_source_dir_path) =
-            paths_for_package(&scarb_metadata, &package.id)?;
-        env::set_current_dir(package_path.clone())?;
-
-        // TODO(#671)
-        let target_dir = target_dir_for_package(&scarb_metadata.workspace.root)?;
-
-        let build_output = Command::new("scarb")
-            .arg("build")
-            .stderr(Stdio::inherit())
-            .stdout(Stdio::inherit())
-            .output()
-            .context("Failed to build contracts with Scarb")?;
-        if !build_output.status.success() {
-            bail!("Scarb build did not succeed")
+    let rt = Runtime::new()?;
+    let all_failed_tests = rt.block_on(async {
```

Comment:
Any reason we're creating an runtime on our own here instead of `tokio::main`? Just to be more explicit?

Reply:
https://github.com/foundry-rs/starknet-foundry/pull/775#discussion_r1345516498 -
It was @MaksymilianDemitraszek suggestion